stdlib/LinearAlgebra/test/cholesky.jl

```diff
@@ -548,4 +548,11 @@ end
     @test logabsdet(B)[1] == -Inf
  end
 
+@testset "diag" begin
+    A = randn(100, 100)
+    P = A' * A
+    C = cholesky(P)
+    @test diag(P) ≈ diag(C)
```

Comment:
```suggestion
    for T in (Float64, ComplexF64), k in (0, 1, -3), uplo in (:U, :L)
        A = randn(T, 100, 100)
        P = Hermitian(A' * A, uplo)
        C = cholesky(P)
        @test diag(P, k) ≈ diag(C, k)
    end
```

Reply:
Incredible! Thank you :)

Reply:
I'll fix this when I get into work today.